cpp/src/plasma/external_store_worker.h

```diff
@@ -0,0 +1,158 @@
+#ifndef EXTERNAL_STORE_WORKER_H
+#define EXTERNAL_STORE_WORKER_H
+
+#include <unistd.h>
+
+#include <condition_variable>
+#include <future>
+#include <iostream>
+#include <mutex>
+#include <queue>
+#include <thread>
+
+#include "plasma/common.h"
+#include "plasma/external_store.h"
+
+namespace plasma {
+
+// ==== The external store worker ====
+//
+// The worker maintains a thread-pool internally for servicing Get requests.
```

Comment:
Arrow already has a thread pool, why not use it?

Reply:
I'm using my thread-pool in some ways that the Arrow thread-pool doesn't support:
1. I'm maintaining a queue of object IDs to fetch, and fetching multiple object IDs per thread to improve performance via batching.
2. I'm using the thread-pool to detect if too many Get() requests have been enqueued, and send an error message back to the caller so that the calling thread can try a synchronous Get() instead.

Reply:
Please add this as a comment in the code, so that future readers understand the choice.

Reply:
Sure, I'll add a comment.